docs/topics/proc_exposing-application-metrics-with-prometheus-spring-boot.adoc

```diff
@@ -0,0 +1,165 @@
++
+--
+Start Prometheus and wait until the `Server is ready to receive web requests` message is displayed in the console.
+
+[source,bash,options="nowrap"]
+----
+$ ./prometheus
+----
+--
+
+.Procedure
+
+. Include the `spring-boot-starter-actuator`, `micrometer-registry-prometheus` and `micrometer-spring-legacy` dependencies in your `pom.xml`.
+The `micrometer-spring-legacy` dependency is required to provide Micrometer support for {SpringBoot} 1.5 applications.
++
+.pom.xml
+[source,xml]
+----
+<dependencies>
+  ...
+  <dependency>
+    <groupId>org.springframework.boot</groupId>
+    <artifactId>spring-boot-starter-actuator</artifactId>
+  </dependency>
+  <dependency>
+    <groupId>io.micrometer</groupId>
+    <artifactId>micrometer-spring-legacy</artifactId>
+  </dependency>
+  <dependency>
+    <groupId>io.micrometer</groupId>
+    <artifactId>micrometer-registry-prometheus</artifactId>
+  </dependency>
+  ...
+</dependencies>
+----
++
+`spring-boot-starter-actuator` automatically records a `Timer` metric on each endpoint exposed your application.
+The default name for the `Timer` metric is set to `http.server.requests`.
+You can set a custom name for the metric:
++
+[source,properties,options="nowrap"]
+----
+# default is `http.server.requests`
+management.metrics.web.server.requests-metric-name=custom.timer.metric.name
+----
++
+The `Timer` metric contains a set of dimensions recorded for every request.
+By default, these dimensions are:
++
+`method`:: the HTTP method used when calling the endpoint (for example, `GET` or `PUT`)
+`status`:: the numeric HTTP status code returned in response to the request (for example, `200`, `201`, `500`)
+`uri`:: the URI template prior to variable substitution (for example, `/api/person/{id}`)
+`exception`:: if an exception is thrown by your application upon receiving a request, this dimension records the simple name of the exception class
+
+. Alternatively, to only record the `Timer` metric for specifics endpoints:
++
+--
+. Disable the `auto-time-requests` property in the `spring-boot-starter-actuator` configuration for your application.
++
+[source,properties,options="nowrap"]
+----
+# true by default
+management.metrics.web.server.auto-time-requests=false
+----
+
+. Annotate methods or classes with the `@io.micrometer.core.annotation.Timed` annotation, for example:
++
+[source,java,options="nowrap"]
+----
+@RestController
+public class GreetingController {
+
+    @RequestMapping("/greeting")
+    @Timed
+    public String get() {
+        return "Hello from a timed endpoint";
+    }
+}
+----
++
+The `@Timed` annotation is used to keep track of the time it takes to invoke the metered endpoint.
+--
+
+. Launch your application:
++
+[source,bash,opts="nowrap"]
+----
+$ mvn spring-boot:run
+----
+
+. Invoke the traced endpoint several times:
++
+[source,bash,opts="nowrap"]
+----
+$ curl http://localhost:8080/greeting
+{"id":1,"content":"Hello from a timed endpoint"}
```

Comment:
The correct output is: `Hello from a timed endpoint`

Reply:
@geoand I see, so the out put is not displayed as JSON. Let me fix that.